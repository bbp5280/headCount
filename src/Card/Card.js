import React from 'react';
import styles from '../index.css';
import PropTypes from 'prop-types';

const Card = ({ location, data, compare, forCompare }) => {
  const yearArray = Object.keys(data);

  const mappedYears = yearArray.map(year => {
    return (
      <div className="info-container" key={year}>
        <span className="data">{data[year]}</span>
        <span className="year">{year}</span>
      </div>
    )
  });

  const clickHandler = () => {
    compare(location);
  }

  const buildCards = (year, click) => {

    if(forCompare === true){
      return(
      <div>
        <div className="top-of-card">
          <h3 className="district">{location}</h3>
          <button className="compare-btn" onClick={click}>Remove
          </button>
        </div>
        <ul className="list">
          {year}
        </ul>
      </div>)
  } else {
    return(
      <div>
      <div className="top-of-card">
        <h3 className="district">{location}</h3>
        <button className="compare-btn" onClick={click}>Compare
        </button>
      </div>
      <ul className="list">
        {year}
      </ul>
    </div>
  )}

  }

  return (

    <div className="card">
      {buildCards(mappedYears, clickHandler)}
    </div>
  )

Card.propTypes = {
  location: PropTypes.string.isRequired,
  data: PropTypes.objectOf(PropTypes.number).isRequired,
  compare: PropTypes.func.isRequired
 }
}

export default Card;
