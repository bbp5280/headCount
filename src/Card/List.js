import React from 'react';
import styles from '../index.css';
import PropTypes from 'prop-types';

const List = ({data}) => {
  const yearArray = Object.keys(data)


  return (
    <div>
      {
        yearArray.map(year => {
        return (
          <div className="info-container" key={year}>
            <span className="data">{data[year]}</span>
            <span className="year">{year}</span>
          </div>
        )
        })
        }
    </div>
  )
};


export default List;
