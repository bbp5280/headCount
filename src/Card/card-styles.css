@import url(https://fonts.googleapis.com/css?family=Lato);
* {
  font-family: 'Lato', sans-serif; }

.card {
  border: 1px solid #e4e4e4;
  background-color: #fff;
  width: 75%;
  margin: 0 auto;
  margin-top: 25px;
  border-radius: 5px;
  padding: 5px;
  padding-top: 17px; }

.top-of-card {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  width: 100%;
  padding-bottom: 10px; }

.district {
  color: #e64f40;
  width: 35%;
  margin-left: 12px;
  font-size: 120%; }

.compare-btn {
  border: none;
  background-color: #F38C73;
  border-radius: 3px;
  margin-right: 12px;
  height: 10%;
  width: 15%;
  font-size: 90%;
  outline-color: #F38C73; }

.compare-btn-off {
  border: none;
  background-color: #829595;
  border-radius: 3px;
  margin-right: 12px;
  height: 10%;
  width: 15%;
  font-size: 90%;
  outline-color: #829595; }

.compare-btn:hover {
  background-color: #e64f40;
  cursor: pointer; }

.toggle-compare-btn {
  background-color: #f3f3f3; }

.info-container {
  display: flex;
  flex-direction: column;
  width: 83px;
  text-align: center; }

.list {
  display: flex;
  padding: 12px; }

.data {
  padding-bottom: 7px; }

.year {
  font-weight: bold; }
