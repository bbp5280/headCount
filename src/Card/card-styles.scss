@import url(https://fonts.googleapis.com/css?family=Lato);

* {
  font-family: 'Lato', sans-serif;
}

.card {
  border: 1px solid #e4e4e4;
  background-color: #fff;
  width: 75%;
  margin: 0 auto;
  margin-top: 25px;
  border-radius: 5px;
  padding: 5px;
  padding-top: 17px;
}

.top-of-card {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  width: 100%;
  padding-bottom: 10px;
}

.district {
  color: #e64f40;
  width: 35%;
  margin-left: 12px;
  font-size: 120%;
}

.compare-btn {
  border: none;
  background-color: #e64f40;
  border-radius: 3px;
  margin-right: 12px;
  height: 10%;
  width: 15%;
  font-size: 90%;
  outline-color: #e64f40;
  color: #fff;
  text-transform: uppercase;
  padding: 4px;
}

.compare-btn-off {
  border: none;
  background-color: #829595;
  border-radius: 3px;
  margin-right: 12px;
  height: 10%;
  width: 15%;
  font-size: 90%;
  outline-color: #829595;
}

.compare-btn:hover {
  background-color: #F38C73;
  cursor: pointer;
}

.toggle-compare-btn {
  background-color: #f3f3f3;
}

.list {
  display: flex;
  flex-direction: row;
  padding: 12px;
}

.list div {
  display: flex;
  flex-direction: row;
}

.info-container {
  //display: inline-block;
  display: flex;
  flex-direction: column;
  width: 83px;
  text-align: center;
}

.data {
  padding-bottom: 7px;
  color: #AC3B30;
}

.year {
  font-weight: bold;
}

@media (max-width: 950px) {

  .card {
    width: 85%;
  }

}

@media (max-width: 650px) {

  .card {
    width: 90%;
  }

}
