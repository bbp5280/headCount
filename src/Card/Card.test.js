import React from 'react';
import Card from './Card';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import config from '../../test/setup.js';

describe('CARD', () => {
  const mkFun = jest.fn();
  let wrapper = shallow(<Card location={'COLORADO'}
                              allData={{2004: 0.24}}
                              key={'COLORADO'}
                              compare={mkFun}
                              />);

it('should match the Card snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
