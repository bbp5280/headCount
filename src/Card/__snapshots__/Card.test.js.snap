exports[`CARD should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    allData={
        Object {
            "2004": 0.24,
          }
    }
    compare={[Function]}
    location="COLORADO" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="top-of-card">
                <h3
                        className="district">
                        COLORADO
                </h3>
                <button
                        className="compare-btn"
                        onClick={[Function]}>
                        Compare
                </button>
        </div>
        <ul
                className="list">
                <List
                        data={
                                Object {
                                        "2004": 0.24,
                                      }
                        } />
        </ul>
</div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="top-of-card">
            <h3
                        className="district">
                        COLORADO
            </h3>
            <button
                        className="compare-btn"
                        onClick={[Function]}>
                        Compare
            </button>
</div>,
          <ul
            className="list">
            <List
                        data={
                                    Object {
                                                "2004": 0.24,
                                              }
                        } />
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="district">
                COLORADO
</h3>,
              <button
                className="compare-btn"
                onClick={[Function]}>
                Compare
</button>,
            ],
            "className": "top-of-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "COLORADO",
                "className": "district",
              },
              "ref": null,
              "rendered": "COLORADO",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Compare",
                "className": "compare-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Compare",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <List
              data={
                            Object {
                                          "2004": 0.24,
                                        }
              } />,
            "className": "list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "2004": 0.24,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="top-of-card">
                    <h3
                              className="district">
                              COLORADO
                    </h3>
                    <button
                              className="compare-btn"
                              onClick={[Function]}>
                              Compare
                    </button>
          </div>
          <ul
                    className="list">
                    <List
                              data={
                                        Object {
                                                  "2004": 0.24,
                                                }
                              } />
          </ul>
</div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="top-of-card">
              <h3
                            className="district">
                            COLORADO
              </h3>
              <button
                            className="compare-btn"
                            onClick={[Function]}>
                            Compare
              </button>
</div>,
            <ul
              className="list">
              <List
                            data={
                                          Object {
                                                        "2004": 0.24,
                                                      }
                            } />
</ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="district">
                  COLORADO
</h3>,
                <button
                  className="compare-btn"
                  onClick={[Function]}>
                  Compare
</button>,
              ],
              "className": "top-of-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "COLORADO",
                  "className": "district",
                },
                "ref": null,
                "rendered": "COLORADO",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Compare",
                  "className": "compare-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Compare",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <List
                data={
                                Object {
                                                "2004": 0.24,
                                              }
                } />,
              "className": "list",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "data": Object {
                  "2004": 0.24,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
